crates/pallet-subspace/src/lib.rs

```diff
@@ -923,9 +926,6 @@ impl<T: Config> Pallet<T> {
             ));
         }
 
-        #[cfg(not(feature = "pot"))]
-        PorRandomness::<T>::take();
```

Comment:
Why shouldn't it be cleared?

Reply:
We should clear it on `on_initialize`, otherwise it only exists during the consensus block execution.

Reply:
It is overridden in `on_initialize`, I do not see why it should be cleared there.

That was actually the goal, to only persist it for the duration of the block execution.

I see it now needs to be called from runtime API, in which case either it needs to be persisted in the other place that derives the randomness or it needs to be renamed here into something more generic. It is not `PoR` randomness and not temporary anymore.

Reply:
Yeah, in this case it should not be temporary anymore.